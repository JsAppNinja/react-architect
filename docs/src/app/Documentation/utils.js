export const splitOnCaps = s => s.split(/(?=[A-Z1-9])/);

function getString(s) {
  return !(s instanceof Array) ? splitOnCaps(s) : s;
}

export const toDashedName = (s) => getString(s).map(s => s.toLowerCase()).join('-');
export const toTitle = (s) => getString(s).join(' ');
