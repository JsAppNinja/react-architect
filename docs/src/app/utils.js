import React from 'react';
export const githubHref = 'https://github.com/mlaursen/react-md';

const lorems = [
  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus at placerat nulla. Morbi eu erat diam. Donec malesuada mauris in neque efficitur luctus. Vestibulum porttitor elit velit, pellentesque imperdiet tellus maximus in. Phasellus consectetur lacus nunc, quis efficitur felis feugiat ac. Vivamus non erat laoreet, volutpat tellus sit amet, porttitor orci. Sed scelerisque id arcu at ornare. Suspendisse potenti. Phasellus eu posuere mi. Fusce venenatis odio dui, ac accumsan lacus dapibus et. Praesent euismod nulla at porta elementum. Nunc ullamcorper velit aliquam velit faucibus tempus.',
  'Duis ligula elit, elementum a nisl a, venenatis fermentum libero. Cras bibendum dignissim nisl, sed sodales tellus scelerisque sed. Sed at enim semper, imperdiet elit et, venenatis magna. Morbi vestibulum purus in ante placerat viverra. Nullam molestie arcu ut dolor rutrum varius. Sed vehicula laoreet volutpat. Quisque feugiat, nisl in sollicitudin consectetur, mi metus aliquet urna, ac sagittis nulla nulla in diam. Fusce id enim auctor ante fringilla auctor in et quam.',
  'Donec pulvinar ligula augue, et tincidunt velit bibendum non. Aenean id metus nunc. Donec non feugiat nisl. In accumsan ligula purus, ut vulputate elit tempus sit amet. Nullam volutpat semper lectus. Nunc a libero tellus. Duis posuere nisi diam, nec euismod risus interdum sit amet. Pellentesque at lacus bibendum, cursus ante non, maximus arcu. Morbi id leo nulla. Integer quis feugiat quam. Etiam auctor pharetra vestibulum.',
  'Nulla sodales metus metus. Sed a metus quis ipsum sagittis placerat. Nunc sit amet molestie purus, nec rutrum quam. Praesent elementum risus eget mattis condimentum. Morbi dignissim, dolor et finibus sollicitudin, nulla enim molestie est, eget tincidunt magna libero et dolor. Fusce tempor sapien et urna sodales, at laoreet nunc egestas. Aenean eget arcu in felis ultrices vehicula. Maecenas interdum viverra sem vitae sodales. Duis tincidunt varius nunc sed viverra. Duis sit amet semper ex, ultrices aliquet mi. Donec a congue sem, ac facilisis ex.',
  'Proin venenatis tortor et ex viverra, ac facilisis mi ullamcorper. Aenean eget est bibendum, fermentum enim eget, semper orci. Nunc magna mauris, fermentum sit amet diam ut, semper malesuada justo. Fusce faucibus tellus eget arcu imperdiet, aliquet placerat massa molestie. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Maecenas gravida quam id ultricies accumsan. Sed porta accumsan libero sed venenatis. Morbi scelerisque, ex in gravida porttitor, ipsum erat sodales urna, mattis mattis nibh ligula quis tellus. Pellentesque consequat et metus scelerisque dapibus. Praesent at maximus ligula. Phasellus vel orci tortor.',
  'Mauris posuere nisl sit amet justo condimentum volutpat. Aliquam venenatis rhoncus est, tristique semper tortor iaculis et. Donec pretium metus nec sodales mollis. Ut vel ipsum sapien. Fusce sit amet libero efficitur orci aliquet scelerisque a ut arcu. In hac habitasse platea dictumst. Sed turpis libero, dictum nec imperdiet nec, maximus eget urna. Sed vel lorem sit amet nisi vulputate mattis. In in odio erat. Aenean euismod lorem nec viverra auctor. Mauris id ipsum maximus, tempus dui nec, mollis felis. Etiam sodales metus sit amet lacus aliquet tristique. Nulla in lacus massa. Morbi quis lorem vitae dolor maximus luctus vel vel eros. Phasellus posuere in purus eget convallis.',
  'Aliquam mauris justo, ornare quis quam fermentum, mollis volutpat dolor. Donec sagittis velit non nisl porttitor, id lacinia tortor vulputate. Suspendisse posuere at felis eu fermentum. Donec at vehicula leo, id sollicitudin ipsum. Donec dapibus sodales erat at malesuada. Mauris sollicitudin velit sed arcu molestie congue. Suspendisse ac arcu enim. Sed ac urna id mi rutrum ornare sit amet in nisl. Phasellus at tempus magna. Nulla facilisi. Mauris pharetra finibus mi ut facilisis. Duis mattis odio id quam vestibulum, id consequat ex fringilla. Aenean sollicitudin accumsan sollicitudin. Duis euismod maximus tempor. Phasellus mi enim, bibendum at nisi vel, cursus venenatis sem. Praesent sagittis, nunc at rhoncus maximus, ligula turpis imperdiet eros, pharetra tincidunt tortor magna id sem.',
  'Sed eleifend sapien condimentum iaculis cursus. In vel pretium turpis. Duis molestie euismod turpis id consectetur. Nunc pharetra sapien libero, a volutpat erat finibus eget. Duis euismod eleifend felis consequat fringilla. Sed varius diam ligula, quis rutrum ex mollis non. Phasellus ac malesuada nibh. Praesent sit amet auctor urna. Quisque eget magna id ex dapibus vulputate eget vel magna.',
  'Ut viverra tempor placerat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras neque diam, auctor eget sapien et, tincidunt blandit turpis. Nam non volutpat velit. Aliquam iaculis euismod nisl non facilisis. Ut blandit dignissim turpis in finibus. Cras neque augue, tempus non neque eu, ornare ultrices sem. Nullam ultricies mi quis semper pellentesque. Mauris mauris lacus, ullamcorper a iaculis quis, tempus id tellus. Phasellus in orci orci. Fusce consequat suscipit ipsum ac bibendum. Maecenas et leo eu leo iaculis varius. Nam consectetur laoreet nisl blandit condimentum.',
  'Ut dictum, sapien vel malesuada posuere, tortor tellus blandit risus, eget hendrerit quam diam eu nibh. Aliquam erat volutpat. Ut rhoncus augue ac nunc porttitor hendrerit. Pellentesque enim risus, auctor quis turpis eget, pulvinar malesuada urna. Nullam tortor mi, elementum id sodales eget, aliquam et nulla. Pellentesque efficitur mi eros, in rutrum metus ultrices vel. Proin vitae diam sit amet nunc pretium malesuada. Fusce at nisi scelerisque, ultrices tortor nec, finibus velit. Donec molestie augue eget leo rutrum, nec ornare arcu gravida. Quisque tincidunt risus elit, ut condimentum eros egestas congue.',
];
export function loremIpsum(paras = 3) {
  let ipsums = [];
  for(let i = 0; i < paras; i++) {
    ipsums.push((
      <p key={i}>{lorems[Math.floor(Math.random() * lorems.length)]}</p>
    ));
  }
  return ipsums;
}

export const numstr = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten'];

export const randomImage = ({ width, height, time } = {}) => {
  width = !width ? 200 : width;
  height = !height ? 200 : height;
  time = typeof time === 'undefined' ? Date.now() : time;
  return `https://unsplash.it/${width}/${height}/?random&time=${time}`;
};

export const hostPrefix = ''; // change to '/react-md' for production
export const imgPrefix = hostPrefix + '/imgs';
