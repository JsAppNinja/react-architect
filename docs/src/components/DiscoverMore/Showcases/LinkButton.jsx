import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { Button } from 'react-md';

const LinkButton = ({ children, className, ...props }) => (
  <Button className={cn('md-cell--right', className)} {...props}>
    {children}
  </Button>
);

LinkButton.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export default LinkButton;
