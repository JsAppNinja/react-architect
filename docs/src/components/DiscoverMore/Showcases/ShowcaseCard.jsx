import React from 'react';
import PropTypes from 'prop-types';
import Button from 'react-md/lib/Buttons/Button';
import Card from 'react-md/lib/Cards/Card';
import CardTitle from 'react-md/lib/Cards/CardTitle';
import CardText from 'react-md/lib/Cards/CardText';
import Media from 'react-md/lib/Media/Media';

const ShowcaseCard = ({ link, logo, name, author, description }) => (
  <Card className="showcase-card" component="li">
    <Media component="a" href={link} title={name}>
      <img src={logo} alt={name} />
    </Media>
    <CardTitle
      title={name}
      subtitle={author.name || author}
    >
      {author.github ? <Button icon iconClassName="fa fa-github" href={author.github} /> : null}
    </CardTitle>
    <CardText>
      {description}
    </CardText>
  </Card>
);

ShowcaseCard.propTypes = {
  link: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  author: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      github: PropTypes.string.isRequired,
    }),
  ]).isRequired,
  description: PropTypes.string.isRequired,
};

export default ShowcaseCard;
