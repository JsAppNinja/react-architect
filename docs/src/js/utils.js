import React from 'react';

import { FontIcon, Avatar } from 'react-md';

export const githubHref = 'https://github.com/mlaursen/react-md';

export const mainLinks = [{
  link: '',
  primaryText: 'Home',
  leftIcon: <FontIcon>home</FontIcon>,
}, {
  link: 'sassdoc',
  primaryText: 'SASS Doc',
  leftAvatar: <Avatar src="/imgs/sass-icon.png" alt="SASS Icon" />,
}];
