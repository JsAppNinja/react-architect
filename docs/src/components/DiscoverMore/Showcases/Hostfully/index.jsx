import React from 'react';

import ShowcaseCard from '../ShowcaseCard';

import logo from './logo.png';

const description = `
Hostfully is a content generation and guest communications platform built for
vacation rental management companies. Using Hostfully, managers upload
property-specific content and local recommendations that travelers can use
while they are at the destination. Companies also use Hostfully as a channel
for guest communication – instead of overloading guests with too many details
about the property, managers direct their guests to well-structured content in
each property’s customized online guidebook – accessible on mobile, desktop, and
in print.
`;

const author = {
  name: 'Hostfully',
  github: 'https://github.com/hostfully',
};

const Hostfully = props => (
  <ShowcaseCard
    {...props}
    name="Hostfully"
    link="https://hostfully.com"
    demoLink="https://v2.hostfully.com/california-dreaming"
    logo={logo}
    author={author}
    description={description}
  />
);

export default Hostfully;
