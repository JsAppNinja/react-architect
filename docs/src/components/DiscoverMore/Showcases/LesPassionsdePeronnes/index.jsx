import React from 'react';

import ShowcaseCard from '../ShowcaseCard';
import logo from './logo.png';

const description = `
Les Passions de PÃ©ronnes offers you to discover wines of terroirs, wines of
vignerons. Most of them are encountered during oeno-tourist trips, telling
us their wines, their hopes sometimes very crazy.
`;

const author = {
  name: 'Fadi Khadra',
  github: 'https://github.com/fkhadra',
};

const LesPassionsdePeronnes = props => (
  <ShowcaseCard
    {...props}
    name="Les Passions de Peronnes"
    link="https://bredariol.be"
    logo={logo}
    author={author}
    description={description}
  />
);

export default LesPassionsdePeronnes;
