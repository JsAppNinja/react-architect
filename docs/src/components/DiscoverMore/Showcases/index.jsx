import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridList } from 'react-md';

import Markdown from 'components/Markdown';
import withMinHeight from 'components/hoc/withMinHeight';
import LesPassionsdePeronnes from './LesPassionsdePeronnes';
import './_styles.scss';

const markdown = `
## Showcases

The following is a list of products/apps that use react-md other than this documentation site. Their entire
interface might not be implemented with react-md, but it should give a good idea of how it is used.

Want to showcase your app/product? Either open a pull request adding your application or look at
[the github issue](https://github.com/mlaursen/react-md/issues/196) for getting your site added.
`;

const Showcases = ({ style, className }) => (
  <GridList size={12} style={style} className={className}>
    <Markdown markdown={markdown} className="md-text-container" />
    <Grid component="ul" noSpacing className="md-list-unstyled">
      <LesPassionsdePeronnes />
    </Grid>
  </GridList>
);

Showcases.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
};

export default withMinHeight(Showcases);
