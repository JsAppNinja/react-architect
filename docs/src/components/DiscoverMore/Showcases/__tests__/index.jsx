/* eslint-env jest */
import React from 'react';
import { renderRouterSnapshot } from 'utils/testing';

import Showcases from '../';

describe('Showcases', () => {
  it('should render all the showcases', () => {
    const tree = renderRouterSnapshot(<Showcases />);
    expect(tree).toMatchSnapshot();
  });
});
