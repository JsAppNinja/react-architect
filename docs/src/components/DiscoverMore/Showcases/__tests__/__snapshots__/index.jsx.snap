// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcases should render all the showcases 1`] = `
<div
  className="md-grid"
  style={undefined}
>
  <section
    className="markdown-container md-text-container md-cell md-cell--12"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2 id=\\"showcases\\">Showcases</h2>
<p>The following is a list of products/apps that use react-md other than this documentation site. Their entire
interface might not be implemented with react-md, but it should give a good idea of how it is used.</p>
<p>Want to showcase your app/product? Either open a pull request adding your application or look at
<a href=\\"https://github.com/mlaursen/react-md/issues/196\\" class=\\"link\\">the github issue</a> for getting your site added.</p>
",
      }
    }
  />
  <ul
    className="md-grid md-grid--no-spacing md-list-unstyled md-cell md-cell--12"
  >
    <li
      className="md-paper md-paper--1 md-card md-background--card showcase-card"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchStart={[Function]}
    >
      <a
        className="md-media md-media--16-9"
        href="https://bredariol.be/product"
        title="Les Passions de Peronnes"
      >
        <img
          alt="Les Passions de Peronnes"
          src="some static content"
        />
      </a>
      <div
        className="md-card-title md-card-title--primary"
        style={undefined}
      >
        <div
          className="md-card-title--title-block"
        >
          <h2
            className="md-card-title--title md-card-title--large md-text"
            id={undefined}
            tabIndex={null}
          >
            Les Passions de Peronnes
          </h2>
          <h3
            className="md-card-title--title md-text--secondary"
          >
            Fadi Khadra
          </h3>
        </div>
        <a
          className="md-btn md-btn--icon md-pointer--hover md-inline-block"
          disabled={undefined}
          href="https://github.com/fkhadra"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <i
            className="md-icon fa fa-github md-text--inherit"
            style={undefined}
          />
        </a>
      </div>
      <section
        className="md-card-text"
      >
        
Les Passions de PÃ©ronnes offers you to discover wines of terroirs, wines of
vignerons. Most of them are encountered during oeno-tourist trips, telling
us their wines, their hopes sometimes very crazy.

      </section>
    </li>
  </ul>
</div>
`;
